bucket-registry.go

```diff
@@ -9,74 +9,128 @@
 package rosmar
 
 import (
+	"context"
 	"sync"
-
-	"golang.org/x/exp/slices"
 )
 
-// The bucket registry tracks all open Buckets (except in-memory ones) by their URL.
-// This is used by feeds, to forward events from the Bucket that created them to other Buckets on
-// the same file that should also be notified.
-//
-// The registry is not bulletproof; it can be fooled into thinking Buckets on the same file aren't,
-// if their URLs aren't identical. This can happen if:
-//
-// * one of the file paths goes through (different) symlinks
-// * the paths are capitalized differently, on a case-preserving filesystem like macOS or Windows
-// * there's some kind of character-encoding difference that the filesystem ignores
+// The bucket registry tracks all open Buckets and refcounts them.
+// The canonical version of the bucket is in the buckets member of the bucketRegistry. This will remain in memory for the following length of time:
 //
-// The way to fix this is to ask the filesystem to canonicalize the path (e.g. by calling
-// `realpath` or `fcntl(F_GETPATH)`) and use the canonical path as the key.
-// Unfortunately Go doesn't seem to have an API for that.
+// * In Memory bucket: bucket is not deleted until any Bucket's CloseAndDelete is closed.
+// * On disk bucket: bucket is deleted from registry when all there are no open copies of the bucket in memory. Unlike in memory bucket, the bucket will stay persisted on disk to be reopened.
```

Comment:
As discussed, a bit more context on how buckets are managed here will make this easier to maintain,.  In particular I'm thinking of explicitly documenting:
- what's shared between bucket copies (almost everything)
- what's not shared, and why (in particular 'closed')